### ISSUE TITLE
mp3/ogg file support instead of midi instruments

### ISSUE THREAD

**Issue by rwtnb** [rwtnb/Drumsthesia#3]
Playing with the actual song mp3 instead of midi instruments? You then could even use one of those apps/libraries that split a song into different tracks using magic and mute the drums? Syncing notes to audio might be an issue there though.

**Comment by Punkoivan** [comment]
To sync the notes we can test next case:
1. Load mp3/ogg
2. Convert it to midi
3. Play mp3 while use midi for notes

The problem is in point 2 - mp3/ogg to midi is not easy thing and would require few tool.
Like lalala.ai to extract drums and then another tool to convert drum to midi.
I've used few online solution to convert mp3 to midi and with vocal the result is garbage. So need to remove vocal first I guess.

**Comment by rwtnb** [comment]
I was thinking about: https://github.com/deezer/spleeter

Audio quality must not be a big issue, since most people would prefer to have the drums louder than the backing track, I think.
But I'm don't like the idea of implementing it inside Drumsthesia, besides it is written in python it would be an ugly solution trying to integrate both.
It would be way cooler if it was implemented as a webapp, where people could share pre-synced midi/drumless audio files, must think about if there is any legal issues here. 
But if we don't store the files it should be ok, until we figure it out.

Basically, the webapp would:
- Split the tracks
- Merge it again excluding the drums
- Use the isolated drum track to predict when the drum part starts
- Try to match it automatically against the user provided midi track, but giving the user an option to fine tune it
- Export a GZIP file with both mid/mp3 plus a metadata file with timing data.

In Drumsthesia, we would just check the file type and play it.

**Comment by rwtnb** [comment]
You know what, a simple CLI would do for now.
